offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/dataresources/projects/ProjectDbConnector.groovy

```diff
@@ -0,0 +1,162 @@
+        addPersonToProject(it, projectID, customerID, "PI")
+      
+        it.commit()
+        
+      } catch (Exception e) {
+        log.error(e.message)
+        log.error(e.stackTrace.join("\n"))
+        it.rollback()
+        it.close() //is this needed?
+        throw new DatabaseQueryException("Could not add person and project data to user database.")
+      }
+    }
+  return new Project(projectIdentifier, projectTitle, projectApplication.getLinkedOffer())
+ }
+
+  private int isProjectInDB(String projectIdentifier) {
+    log.info("Looking for project " + projectIdentifier + " in the DB");
+    String sql = "SELECT * from projects WHERE openbis_project_identifier = ?";
+    int res = -1;
+    Connection connection = connectionProvider.connect()
+    connection.withCloseable { it ->
+      PreparedStatement statement = it.prepareStatement(sql);
+      statement.setString(1, projectIdentifier);
+      ResultSet rs = statement.executeQuery();
+      if (rs.next()) {
+        res = rs.getInt("id");
+      }
+    }
+    return res;
+  }
+
+  public int addProjectToDB(Connection connection, String projectIdentifier, String projectName) {
+    int exists = isProjectInDB(projectIdentifier);
+    if (exists < 0) {
+      log.info("Trying to add project " + projectIdentifier + " to the person DB");
+      String sql = "INSERT INTO projects (openbis_project_identifier, short_title) VALUES(?, ?)";
+      try (PreparedStatement statement =
+      connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS)) {
+        statement.setString(1, projectIdentifier);
+        statement.setString(2, projectName);
+        statement.execute();
+        ResultSet rs = statement.getGeneratedKeys();
+        if (rs.next()) {
+          logout(conn);
+          log.info("Successful.");
+          return rs.getInt(1);
+        }
+      } catch (Exception e) {
+        log.error("SQL operation unsuccessful: " + e.getMessage());
+        e.printStackTrace();
+      }
+      return -1;
+    }
+    return exists;
+  }
+
+  public void addPersonToProject(Connection connection, int projectID, int personID, String role) {
+    if (!hasPersonRoleInProject(personID, projectID, role)) {
+      log.info("Trying to add person with role " + role + " to a project.");
+      String sql =
+      "INSERT INTO projects_persons (project_id, person_id, project_role) VALUES(?, ?, ?)";
+      try (PreparedStatement statement =
+      connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS)) {
+        statement.setInt(1, projectID);
+        statement.setInt(2, personID);
+        statement.setString(3, role);
+        statement.execute();
+        log.info("Successful.");
+      } catch (Exception e) {
+        log.error("SQL operation unsuccessful: " + e.getMessage());
+        e.printStackTrace();
+      }
+    }
+  }
+
+  public boolean hasPersonRoleInProject(int personID, int projectID, String role) {
```

Comment:
JavaDocs please :)

Reply:
I am not sure, why you need to ask for a role? Why is this important in the context of project creation based on the offer?

Reply:
in the junction table one person can have multiple roles for a project. here I test if a person already has that role for that project so they don't need to be added. this is code from the projectwizard

Reply:
👍  Thank you for elaborating